test/hotspot/jtreg/compiler/arguments/TestPrintOptoAssemblyLineNumbers.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8033441
+ * @summary Test to ensure that line numbers are now present with the -XX:+PrintOptoAssembly command line option
+ *
+ * @requires vm.debug == true
```

Comment:
the flag `PrintOptoAssembly` comes from c2. 
IMHO, you need `@requires vm.compiler2.enabled` instead of vm.debug == true.

Reply:
Thanks for raising this, I found that without `vm.debug == true` the test will fail if run on a non debug build (as the expected output is only available on the debug build). But I have adjusted the `@requires` to include `vm.compiler2.enabled`